### PR TITLE
Add option to evaluate every next item in next-item evaluation

### PR DESCRIPTION
**Pull request by lthoang**
### Related Issues

https://github.com/PreferredAI/cornac/issues/578

### Checklist:

- [x] I have added tests.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated `README.md` (if you are adding a new model).
- [ ] I have updated `examples/README.md` (if you are adding a new example).
- [ ] I have updated `datasets/README.md` (if you are adding a new dataset).

### CODE REVIEW

File: cornac/eval_methods/base_method.py [hunk 1]
```diff
@@ -272,6 +272,7 @@ def __init__(
         self.val_set = None
         self.rating_threshold = rating_threshold
         self.exclude_unknowns = exclude_unknowns
+        self.mode = kwargs.get("mode", None)
```
Comment (tqtg) [inline cornac/eval_methods/base_method.py:275]: If `mode` is only used for next-item evaluation, I recommend to move it out of the base method

File: cornac/eval_methods/next_item_evaluation.py [hunk 2]
```diff
@@ -23,13 +23,19 @@
 from . import BaseMethod
 
 
+EVALUATION_MODES = [
```
Comment (tqtg) [inline cornac/eval_methods/next_item_evaluation.py:26]: Let's use set instead of list

File: cornac/eval_methods/next_item_evaluation.py [hunk 3]
```diff
@@ -191,8 +210,11 @@ def __init__(
             seed=seed,
             exclude_unknowns=exclude_unknowns,
             verbose=verbose,
+            mode=mode,
             **kwargs,
         )
+        assert mode in EVALUATION_MODES
```
Comment (tqtg) [inline cornac/eval_methods/next_item_evaluation.py:216]: Please raise an exception and provide informative message with the supported eval modes. This argument is quite critical.

File: cornac/eval_methods/base_method.py [hunk 4]
```diff
@@ -754,6 +756,7 @@ def evaluate(self, model, metrics, user_based, show_validation=True):
             rating_metrics=rating_metrics,
             ranking_metrics=ranking_metrics,
             user_based=user_based,
+            mode=self.mode,
```
Comment (tqtg) [inline cornac/eval_methods/base_method.py:759]: We probably don't need to provide `mode` here, it's only for next-item eval.

**Review by tqtg — CHANGES_REQUESTED** [review]

File: cornac/eval_methods/base_method.py (hunk 1 above)
Comment (lthoang) [reply cornac/eval_methods/base_method.py]: Updated

**Review by tqtg — APPROVED** [review]

**Comment by tqtg** [comment]
LGTM. Feel free to merge

**Pull request merged**